### ISSUE TITLE
关于卷积层的一点疑问

### ISSUE THREAD

**Issue by YaoHuUstc** [li-ziang/ProtCNN#1]
您好，请问在process_data.py程序中和run_2048.py程序中residual_block中的卷积层构造为何不一样？是否为process_data.py代码来源错误。祝好

**Comment by li-ziang** [comment]
hi，这是一个好问题！`run_2048.py`中的代码能取得更好的结果，这也是我在README中建议运行`run_2048.py`的原因。`process_data.ipynb`大部分基于[Genetic-ProtCNN](https://github.com/anindya-vedant/Genetic-ProtCNN/blob/master/Notebooks/Pfam_protein_sequence_classification_full.ipynb)，是用来参考如何处理数据的，同时可以作为一个baseline，因为其训练结果并不是很理想。

**Comment by YaoHuUstc** [comment]
感谢您的回复。我还有一点疑问。process_data.ipynb中的conv1 = Conv1D(filters, 1, dilation_rate=d_rate, padding='same', kernel_regularizer=l2(0.001))(act1)。如果conv1d的kernelsize设为1.那么dilation_rate的意义是什么？是否意味着和stride是一样的效果，只不过stride不会进行padding。这是否是[Genetic-ProtCNN]出现的错误，还是有意为之。非常感谢您，希望您能解答一下我的疑惑。

**Comment by li-ziang** [comment]
介于我很长时间没有使用TensorFlow了，我不敢轻易下结论。所以我建议你在[TensorFlow Conv1d文档](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv1D)中学习每个参数的意义。如果有条件的话可以自己用模拟一个输入试试。

祝好！

**Comment by YaoHuUstc** [comment]
好的，非常感谢您的回复。
